### ISSUE TITLE
Question: how to continue the training with more or fewer GPUs

### ISSUE THREAD

**Issue by amsword** [microsoft/DeepSpeed#1721]
If there are N GPUs, the snapshot will be N files for optimizer states. Each file corresponds to 1 GPU. (let me know if the understanding is not correct). Then, how to continue the training with more GPU, say, 2N GPUs? Is there an easy way to consolidate the optimizer states?

**Comment by tjruwase** [comment]
Yes, the N optimizer state files correspond to the N GPUs. We are working on a feature to support changing the number of GPUs between training runs. Can you try setting the following parameter to true in your zero config?
<img width="290" alt="image" src="https://user-images.githubusercontent.com/4271600/150864781-c41188ee-c300-4fc0-947b-1aa9140b1576.png">

**Comment by itsnamgyu** [comment]
Are there any resources on how to do this manually?

**Comment by tjruwase** [comment]
@itsnamgyu, please see in-development feature called Universal Checkpointing
https://github.com/microsoft/Megatron-DeepSpeed/blob/main/examples_deepspeed/universal_checkpointing/README.md
